src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -118,6 +132,61 @@ public void kill(Map<String, String> modelEnvVars) throws IOException, Interrupt
                 getListener().getLogger().println("Killing process.");
                 ContainerExecDecorator.this.close();
             }
+
+
+            private boolean isContainerReady(Pod pod, String container) {
+                if (pod == null || pod.getStatus() == null || pod.getStatus().getContainerStatuses() == null) {
+                    return false;
+                }
+
+                for (ContainerStatus info : pod.getStatus().getContainerStatuses()) {
+                    if (info.getName().equals(container) && info.getReady()) {
+                        return true;
+                    }
+                }
+                return false;
+            }
+
+            private boolean waitUntilContainerIsReady() {
+                final Pod pod = client.pods().withName(podName).get();
+
+                if (pod == null) {
+                    throw new IllegalArgumentException("Container with name:[" + containerName+"] not found in pod:[" + podName + "]");
+                }
+                if (isContainerReady(pod, containerName)) {
+                    return true;
+                }
+
+                launcher.getListener().getLogger().println("Waiting for container container [" + containerName + "] of pod [" + podName + "] to become ready.");
+                final CountDownLatch latch = new CountDownLatch(1);
+                Watcher<Pod> podWatcher = new Watcher<Pod>() {
+                    @Override
+                    public void eventReceived(Action action, Pod resource) {
+                        switch (action) {
+                            case MODIFIED:
+                                if (isContainerReady(pod, containerName)) {
+                                    latch.countDown();
+                                }
+                                break;
+                            default:
+                        }
+                    }
+
+                    @Override
+                    public void onClose(KubernetesClientException cause) {
+
+                    }
+                };
+
+                try (Watch watch = client.pods().withName(podName).watch(podWatcher)) {
+                    if (latch.await(5, TimeUnit.MINUTES)) {
```

Comment:
Can you put the 5 in a `System.getProperty(ContainerExecDecorator.class.getName()+".containerReadyTimeout","5");` ?